src/resizer/resizer.js

```diff
@@ -0,0 +1,138 @@
+            resizing: "resizer-resizing",
+        };
+        this.mouseDownHandler = (event) => this._onMouseDown(event);
+    }
+
+    setClassNames(classNames) {
+        this.classNames = classNames;
+    }
+
+    attach() {
+        this.container.addEventListener("mousedown", this.mouseDownHandler, false);
+    }
+
+    detach() {
+        this.container.removeEventListener("mousedown", this.mouseDownHandler, false);
+    }
+
+    /**
+    Gives the distributor for a specific resize handle, as if you would have started
+    to drag that handle. Can be used to manipulate the size of an item programmatically.
+    @param {number} handleIndex the index of the resize handle in the container
+    @return {Distributor} a new distributor for the given handle
+    */
+    forHandleAt(handleIndex) {
+        const handles = this._getResizeHandles();
+        const handle = handles[handleIndex];
+        const {distributor} = this._createSizerAndDistributor(handle);
+        return distributor;
+    }
+
+    _isResizeHandle(el) {
+        return el && el.classList.contains(this.classNames.handle);
+    }
+
+    _onMouseDown(event) {
+        const target = event.target;
+        if (!this._isResizeHandle(target) || target.parentElement !== this.container) {
+            return;
+        }
+        // prevent starting a drag operation
+        event.preventDefault();
+        // mark as currently resizing
+        if (this.classNames.resizing) {
+            this.container.classList.add(this.classNames.resizing);
+        }
+
+        const {sizer, distributor} = this._createSizerAndDistributor(target);
+
+        const onMouseMove = (event) => {
+            const offset = sizer.offsetFromEvent(event);
+            distributor.resize(offset);
+        };
+
+        const body = document.body;
+        const onMouseUp = (event) => {
+            if (this.classNames.resizing) {
+                this.container.classList.remove(this.classNames.resizing);
+            }
+            body.removeEventListener("mouseup", onMouseUp, false);
+            body.removeEventListener("mousemove", onMouseMove, false);
+        };
+        body.addEventListener("mouseup", onMouseUp, false);
+        body.addEventListener("mousemove", onMouseMove, false);
+    }
+
+    _createSizerAndDistributor(resizeHandle) {
+        const vertical = resizeHandle.classList.contains(this.classNames.vertical);
+        const reverse = resizeHandle.classList.contains(this.classNames.reverse);
+
+        // eslint-disable-next-line new-cap
+        const sizer = new this.sizerCtor(this.container, vertical, reverse);
+
+        const items = this._getResizableItems();
+        const prevItem = resizeHandle.previousElementSibling;
+        // if reverse, resize the item after the handle instead of before, so + 1
+        const itemIndex = items.indexOf(prevItem) + (reverse ? 1 : 0);
+        const item = items[itemIndex];
+        // eslint-disable-next-line new-cap
```

Comment:
why are we disabling the lint new line warning?

Reply:
`new-cap` is the warning that constructors should be capitalized. As the constructor is a member variable, I thought it made more sense to follow the convention for member variables.